terraform/modules/ecs/ecs.tf

```diff
@@ -0,0 +1,231 @@
+#tfsec:ignore:aws-ecs-enable-container-insight:TODO CLDC-2542 enable container insights if necessary for logging/monitoring
+resource "aws_ecs_cluster" "main" {
+  #checkov:skip=CKV_AWS_65:TODO CLDC-2542 enable container insights if necessary for logging/monitoring
+  name = "${var.prefix}-ecs-cluster"
+}
+
+resource "aws_ecs_task_definition" "main" {
+  #checkov:skip=CKV_AWS_336:using readonlyRootFilesystem to true breaks the app, as it needs to write to app/tmp/pids for example
+  family                   = "${var.prefix}-ecs-task"
+  cpu                      = var.ecs_task_cpu
+  execution_role_arn       = aws_iam_role.task_execution.arn
+  memory                   = var.ecs_task_memory #MiB
+  network_mode             = "awsvpc"
+  requires_compatibilities = ["FARGATE"]
+  task_role_arn            = aws_iam_role.task.arn
+
+  container_definitions = jsonencode([
+    {
+      name              = "${var.prefix}-ecs-container"
+      cpu               = var.ecs_task_cpu
+      environment       = var.ecs_environment_variables
+      essential         = true
+      image             = var.ecr_repository_url
+      memoryReservation = var.ecs_task_memory * 0.75
+      user              = "nonroot"
+
+      logConfiguration = {
+        logDriver = "awslogs"
+        options = {
+          awslogs-group         = aws_cloudwatch_log_group.main.id
+          awslogs-region        = "eu-west-2"
+          awslogs-stream-prefix = var.prefix
+        }
+      }
+
+      portMappings = [
+        {
+          protocol      = "tcp"
+          containerPort = 8080
+          hostPort      = 8080
+        }
+      ]
+
+      secrets = [for key, value in var.ecs_parameters : {
+        name      = key
+        valueFrom = value
+      }]
+    }
+  ])
+
+  runtime_platform {
+    operating_system_family = "LINUX"
+    cpu_architecture        = "X86_64"
+  }
+}
+
+resource "aws_ecs_task_definition" "db_migrate" {

```

Comment:
Quite a bit of code duplication for each one of these task definitions. Mostly the same other than terraform name, `family` and `command` attributes

Reply:
An alternative would be to have just one task definition (for running the app); then whenever we want to manually run something like `db:migrate`, we run the same task definition but overwrite the `command` in the aws console